.github/workflows/main.yml

```diff
@@ -0,0 +1,239 @@
+name: main
+
+on:
+  workflow_call:
+    inputs:
+      # enabled by default, required for merge through default GH check
+      test_ci:
+        description: 'Enable Unit tests'
+        default: true
+        required: false
+        type: boolean
+      # disabled by default, but required for merge, there are two GH checks:
+      # - Non-Application Server integration tests
+      # - Application Server integration tests
+      # opt-in with 'ci:agent-integration'
+      agent_integration_tests_ci:
+        description: 'Enable Agent Integration tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:jdk-compatibility' tag on PR
+      jdk_compatibility_ci:
+        description: 'Enable JDK compatibility tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:windows' tag on PR
+      windows_ci:
+        description: 'Enable Windows build & tests'
+        default: false
+        required: false
+        type: boolean
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+      - ready_for_review
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/maven-goal
+        with:
+          command: ./mvnw install -DskipTests=true -Dmaven.javadoc.skip=true
+      - uses: ./.github/workflows/stash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+
+  license:
+    name: License
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/unstash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+      - name: install artifacts
+        uses: ./.github/workflows/maven-goal
+        with:
+          # Need to install packages in m2 local repository
+          command: ./mvnw install -DskipTests=true -Dmaven.javadoc.skip=true
+      - name: mvn license
```

Comment:
missed `uses: ./.github/workflows/maven-goal`?

Reply:
AFAIS, we need to install project jars in the local maven repository to be able to run the license step.
This step isn't required in most cases.

Reply:
I meant whether `mvn license` should use the reusable action called ` ./.github/workflows/maven-goal`, as far as I see it uses `run: ./mvnw org.codehaus.mojo:license-maven-plugin...`

Reply:
Gotcha, thanks for the explanation! all ✅ in my end!